R/FilterStates.R

```diff
@@ -373,12 +373,20 @@ FilterStates <- R6::R6Class( # nolint
               logger::log_debug("FilterStates$srv_active@2 triggered by added states: { toString(added_state_names) }")
               lapply(added_states(), function(state) {
                 fs_callback <- state$server(id = fs_to_shiny_ns(state))
-                observeEvent(
+                state_id <- state$get_state()$id
+                remove_observer <- observeEvent(
                   once = TRUE, # remove button can be called once, should be destroyed afterwards
                   ignoreInit = TRUE, # ignoreInit: should not matter because we destroy the previous input set of the UI
                   eventExpr = fs_callback(), # when remove button is clicked in the FilterState ui
-                  handlerExpr = private$state_list_remove(state$get_state()$id)
+                  handlerExpr = private$state_list_remove(state_id)
                 )
+
+                # Each filter state_id can only have 1 active remove observer at the time
+                remove_id <- sprintf("remove_%s", state_id)
+                if (!is.null(private$observers[[remove_id]])) {
```

Comment:
This shouldn't be here. If we have one instance of `FilteredData` and multiple `srv_active` (for each module), then it is removing observers from other tabs. For example when `module_specific = FALSE` and two modules use the same filter - it will be possible to remove filter from the last-rendered tab only.

Reply:
We still need to keep track of the observers that are created.

I removed the overwrite, but keeping any additional observers with `remove_<state_id>_<count>`

Any observer that has the state id on its name is removed on method call `state_list_remove` via

```r
state$destroy_observers() # already
lapply(
  Filter(function(x) grepl(state$get_state()$id, x, fixed = TRUE), names(private$observers)),
  function(x) private$observers[[x]]$destroy()
)
```

Ideally we would push the observer to the state$private$observers, but that is private and requires new API on `<teal.slice>/R/FilterState.R`